test/test_evaluators.py

```diff
@@ -838,3 +839,120 @@ def test_l2_evaluator(self):
         evaluator_l2.add_predictions(predictions=self.PREDICTIONS[5:], targets=self.TARGETS[5:])
         report = evaluator_l2.get_report()
         self.assertAlmostEqual(report[evaluator_l2._get_id()], np.sqrt(10) / 10, places=4)
+
+
+class TestInformationRetrievalMetrics(unittest.TestCase):
+    PREDICTIONS = [np.array([[5, 4, 3, 2, 1]]),
```

Comment:
what are the predictions here? Can you explain?

Reply:
Note that the user of this repo just assume it is a generic evalaution metric repo. They have no idea about text-2-image, florence, etc.

MeanAveragePrecisionAtK is currently inheriting from MemorizingEverythingEvaluator, and if the user check the doc of MeanAveragePrecisionAtK.add_predictions, the doc shows up here is:

        """ Add a batch of predictions for evaluation.
        Args:
            predictions: the model output array. Shape (N, num_class)
            targets: the ground truths. Shape (N, num_class) for multi-label or (N,) for multi-class
        """

Reply:
You are absolutely right and I agree with that. This is why I was originally considering implementing separate classes specifically for image retrieval. How do you feel about removing all the special code I added in PrecisionEvaluator, RecallEvaluator and adding new classes PrecisionAtKInformationRetrieval, RecallAtKInformationRetrieval and MeanAveragePrecisionAtKInformationRetrieval with their own doc?

Reply:
alternatively, I can add documentation within the current classes to explain meaning for Image Retrieval

Reply:
That seems overwhelming, you can try refactor the common part out of them with proper naming.